pkg/supportconfig/auth.go

```diff
@@ -0,0 +1,140 @@
+package supportconfig
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	authv1 "k8s.io/api/authentication/v1"
+	v1 "k8s.io/api/authorization/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	authnv1 "k8s.io/client-go/kubernetes/typed/authentication/v1"
+	authzv1 "k8s.io/client-go/kubernetes/typed/authorization/v1"
+)
+
+const (
+	unauthenticatedError = "unauthenticated \n"
```

Comment:
I was testing this endpoint with curl and diskliked the lack of feedback that the endpoint gave (had to use verbose mode to see the response code in the header). 

I can remove this and have it only set the header, I was just trying to make the response include some more info so that it was an easier-to-use api.

Reply:
Resolving due to change of goals with this PR.